apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js

```diff
@@ -25,8 +25,17 @@ function WelcomeTour() {
 		( select ) => {
 			const welcomeGuideStoreSelect = select( 'automattic/wpcom-welcome-guide' );
 			const starterPageLayoutsStoreSelect = select( 'automattic/starter-page-layouts' );
+			const isFetchingSiteDetails = Boolean( select( 'automattic/site' ).isFetchingSiteDetails() );
+			// default isP2 to true when still loading, because we don't want to show the tour for P2s
+			// this makes sure the tour doesn't appear for a second until we find out the site is a P2
+			const isP2 = Boolean(
+				isFetchingSiteDetails ||
+					select( 'automattic/site' ).getSite( window._currentSiteId )?.options?.is_wpforteams_site
+			);
 			return {
-				show: welcomeGuideStoreSelect.isWelcomeGuideShown(),
+				// hide welcome tour for P2s
+				// see: p1650559200031119-slack-C025Q5RK2.
```

Comment:
GH issue is prolly better reference: https://github.com/Automattic/wp-calypso/issues/62973